train.py

```diff
@@ -135,23 +140,22 @@ def train_epoch(
             for p in model.parameters():
                 param_norm = p.grad.detach().data.norm(2)
                 total_norm += param_norm.item() ** 2
-            b += 1 if total_norm ** 0.5 < S else 0
+            b += 1 if total_norm ** 0.5 < S_e else 0
```

Comment:
I think we should apply `S_e` (the value that was updated after the last batch), right? This also applies to 146. Another option is to use `S` from the previous epoch.

Reply:
Right